File: Java/Bridge/ColorfulShapes/src/com/ericgtkb/Color.java
package com.ericgtkb;

public interface Color {
    void fillColor();
}


class RedColor implements Color {
    @Override
    public void fillColor() {
        System.out.println("Filling the space with red...");
    }
}


class GreenColor implements Color {
    @Override
    public void fillColor() {
        System.out.println("Filling the space with green...");
    }
}


class BlueColor implements Color {
    @Override
    public void fillColor() {
        System.out.println("Filling the space with blue...");
    }
}

File: C++/Bridge/ColorfulShapes/color.cc
#include <iostream>
#include "color.h"

void RedColor::fill_color() const {
    std::cout << "Filling the space with color red...\n";
}


void GreenColor::fill_color() const {
    std::cout << "Filling the space with color green...\n";
}


void BlueColor::fill_color() const {
    std::cout << "Filling the space with color blue...\n";
}
